frontend/tests/pages/search/pages.test.tsx

```diff
@@ -0,0 +1,84 @@
+import { render, screen } from "@testing-library/react";
+import Search from "src/app/[locale]/search/page";
+import { MockSearchFetcher } from "src/services/search/searchfetcher/MockSearchFetcher";
+
+function mockUseTranslations(translationKey: string) {
+  return translationKey;
+}
+
+mockUseTranslations.rich = (translationKey: string) => translationKey;
+
+// test without feature flag functionality
+jest.mock("src/hoc/search/withFeatureFlag", () =>
+  jest.fn((Component: React.Component) => Component),
+);
+
+// test without i18n functionality, pass through translation key as text
+jest.mock("next-intl/server", () => ({
+  getTranslations: jest.fn(() => (translationKey: string) => translationKey),
+  unstable_setRequestLocale: jest.fn(),
+}));
+
+jest.mock("next-intl", () => ({
+  useTranslations: () => mockUseTranslations,
+}));
+
+// mock API interactions
+jest.mock("src/services/search/searchfetcher/SearchFetcherUtil", () => ({
+  getSearchFetcher: () => new MockSearchFetcher(),
+}));
+
+jest.mock("next/navigation", () => ({
+  ...jest.requireActual<typeof import("next/navigation")>("next/navigation"),
+  useRouter: () => ({
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    push: () => {},
+  }),
+}));
+
+/*
+  nested async server components (< ...Fetcher />) are currently breaking the render.
+  stated workarounds are not working. to get testing minimally working, overriding
+  Suspense to force display of fallback UI.
+
+  for more see https://github.com/testing-library/react-testing-library/issues/1209
+*/
+jest.mock("react", () => ({
+  ...jest.requireActual<typeof import("react")>("react"),
+  Suspense: ({ fallback }: { fallback: React.Component }) => fallback,
+}));
+
+describe("Search Route", () => {
+  it("renders the search page with expected checkboxes checked", async () => {
+    const mockSearchParams = {
+      status: "forecasted,posted",
+    };
+
+    render(<Search searchParams={mockSearchParams} />);
```

Comment:
We could add more scenarios to test sorting, paging, input component etc in a later ticket or PR.

Reply:
maybe? the potential issue with all the mocking I've done to get this working is that the dynamic behavior we get from things like `useRouter` or API requests are getting lost at the moment. I can play around with what's possible, but I am curious as to the team's philosophy regarding the purpose of component unit tests vs. route level unit (?) tests vs. end to end integration tests and what level of functionality we want to test where. In my mind it could be a better use of time to thoroughly test things in the lower level components and integration tests - which it looks like we already have - and zeroing in on what the purpose of the page level tests like this are doing that is not captured elsewhere